core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/common/WorkflowActor.scala

```diff
@@ -28,11 +28,11 @@ abstract class WorkflowActor(val identifier: ActorVirtualIdentity) extends Actor
   )
   lazy val controlInputPort: ControlInputPort = wire[ControlInputPort]
   lazy val controlOutputPort: ControlOutputPort = wire[ControlOutputPort]
-  lazy val promiseManager: PromiseManager = wire[PromiseManager]
-  val promiseHandlerInitializer: PromiseHandlerInitializer
-
-  // register all handlers to the promise manager
-  promiseManager.setPromiseHandlers(promiseHandlerInitializer.getPromiseHandlers)
+  lazy val rpcClient: RPCClient = wire[RPCClient]
+  lazy val rpcServer: RPCServer = wire[RPCServer]
+  // this variable cannot be lazy
+  // because it should be initialized with the actor itself
+  val rpcHandlerInitializer: RPCHandlerInitializer
```

Comment:
I may be wrong here but I cannot understand. Arent handlers used inside RPCServers only? So if server is lazy shouldnt handlers be too?

Reply:
No, HandlerInitializer will modify the handlers inside RPCServer. So initializing HandlerInitializer will also initialize RPCServer on construction. If we mark HandlerInitializer as lazy, they will both be initialized when the first control message comes, which will slow down the speed of handling that control message.

Reply:
got it